### PR TITLE
Update docker file to use ubi-minimal for the build phase

### PR DESCRIPTION
**Pull request by xrajesh**
This is to address https://github.com/kruize/autotune/issues/613

### CODE REVIEW

File: Dockerfile.autotune [hunk 1]
```diff
@@ -16,15 +16,21 @@
...
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends git vim \
-    && rm -rf /var/lib/apt/lists/*
+RUN microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install -y java-17-openjdk-devel \
+     && microdnf clean all \
+     && rpm -q java-17-openjdk-devel
+
+RUN microdnf install tar gzip  java-17-openjdk-jmods binutils
+
+RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
```
Comment (dinogun) [inline Dockerfile.autotune:29]: Please have a single `RUN` that does all of the `microdnf install` and then do the `microdnf clean all` at the end of it.  Also install both `git` and `vim` as those are needed in the build container

File: Dockerfile.autotune [hunk 2]
```diff
@@ -16,15 +16,21 @@
...
 
-# Install any additional packages that a typical developer in the team needs.
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends git vim \
-    && rm -rf /var/lib/apt/lists/*
+RUN microdnf --setopt=install_weak_deps=0 --setopt=tsflags=nodocs install -y java-17-openjdk-devel \
+     && microdnf install tar gzip java-17-openjdk-jmods binutils git vim \
+     && microdnf clean all 
+
+RUN microdnf install tar gzip  java-17-openjdk-jmods binutils
```
Comment (dinogun) [inline Dockerfile.autotune:27]: Thanks for the update @xrajesh, Looks like this line can now be removed?

Comment (xrajesh) [reply Dockerfile.autotune]: Thank you @dinogun , I also updated the Maven version.

Comment (xrajesh) [reply Dockerfile.autotune]: @dinogun Updated maven to 3.9.0 .. Please check.

**Comment by dinogun** [comment]
@xrajesh Will merge this PR, but please GPG sign all your commits as mentioned in the [contributing](https://github.com/kruize/autotune/blob/master/CONTRIBUTING.md) doc.

**Pull request merged**